### PR TITLE
feat(frontend): show local timezone alongside UTC in schedule displays

### PR DESCRIPTION
**Pull request by machadovilaca**
## Summary

Follows up on #1496. Shows the user's local timezone alongside UTC in all schedule-related displays, so users don't have to mentally convert UTC times.

- Add `getCronDescriptionWithLocal()` that inserts "UTC" after the time in cron descriptions and appends the local equivalent in parentheses
- Update schedule column, details card, and form preview to use the new function
- Add "Times are in UTC" helper text to the schedule form
- Sub-daily schedules (every hour, every 15 min) skip the annotation since local offset adds no value

**Before:** `At 05:30 AM, Monday through Friday`
**After:** `At 05:30 AM UTC, Monday through Friday (6:30 AM GMT+1)`

Edit
s1

List
s2

Show
s3

## Why UTC input (not local-first input)

We considered letting users input cron schedules in their local timezone and converting to UTC behind the scenes, but rejected it:

- **DST drift**: `getTimezoneOffset()` returns the offset at input time, not execution time. A user in London (UTC+0 winter, UTC+1 summer) who sets "9 AM local" in January gets `0 9 * * *` UTC — which silently becomes 10 AM local in summer
- **Fractional offsets**: UTC+5:30 (India, 1.4B people), UTC+5:45 (Nepal), UTC+12:45 (Chatham Islands) require shifting both hour and minute fields, with minute overflow creating double-wrap edge cases
- **Day-of-week wrapping**: Timezone shifts crossing midnight require adjusting day-of-week fields — ranges like `1-5` must become `0-4` or `2-6`, with modular wrapping on each element
- **Round-trip conversion**: Editing an existing schedule requires reverse-converting UTC→local, which may produce a different cron than originally entered if DST changed since creation

UTC input avoids all of these. Showing the local equivalent next to it gives users the "when does this fire for me?" answer without any conversion complexity or correctness risks.

## Test plan

- [x] Unit tests for `getCronDescriptionWithLocal()` — daily, weekly, monthly schedules include UTC + local; sub-daily schedules return plain description; invalid cron falls back gracefully
- [x] All frontend unit tests pass
- [x] Production build passes with 0 errors, 0 warnings
- [x] Visual verification against Kind cluster: schedule list, detail card, and form preview all show dual timezone

## Summary by CodeRabbit

* **New Features**
  * Schedule descriptions now include a UTC label and, when appropriate, a parenthetical local-time preview; sub-daily schedules omit timezone annotations. Table and detail labels simplified to “Schedule”. Schedule preview areas note “Times are in UTC”.

* **Tests**
  * Added tests validating the timezone-aware schedule description behavior and edge cases (sub-daily, invalid cron, 12h/24h formats).

* **Documentation**
  * Added a frontend spec defining timezone display rules for cron schedules.

### CODE REVIEW

File: components/frontend/src/lib/__tests__/cron.test.ts [hunk 1]
```diff
@@ -32,3 +32,53 @@ describe('getNextRuns', () => {
...
+describe('getCronDescriptionWithLocal', () => {
+  it('includes UTC label for daily schedule', () => {
+    const result = getCronDescriptionWithLocal('0 9 * * *');
+    expect(result).toContain('UTC');
+  });
+
+  it('includes local timezone in parentheses for daily schedule when not UTC', () => {
+    const result = getCronDescriptionWithLocal('0 9 * * *');
+    expect(result).toMatch(/UTC/);
+  });
```
[the remark this answers is not in the document]
Comment (machadovilaca) [reply components/frontend/src/lib/__tests__/cron.test.ts]: The test now actually verifies the parenthesized local suffix when the runner isn't in UTC, and correctly expects no parentheses when it is. Skipped the suggested Intl.DateTimeFormat mocking approach, it would couple the test to formatTimeLocal's internals rather than testing the function's observable behavior.

File: components/frontend/src/lib/cron.ts [hunk 2]
```diff
@@ -29,3 +30,31 @@ export function getNextRuns(cronExpr: string, count: number): Date[] {
...
+  }
+
+  const sameDay =
+    runs[0].getUTCFullYear() === runs[1].getUTCFullYear() &&
+    runs[0].getUTCMonth() === runs[1].getUTCMonth() &&
+    runs[0].getUTCDate() === runs[1].getUTCDate();
+
+  if (sameDay) {
+    return description;
+  }
```
[the remark this answers is not in the document]
Comment (machadovilaca) [reply components/frontend/src/lib/cron.ts]: The interval comparison correctly handles boundary cases like 0 */12 * * * regardless of current time.

**Review by jsell-rh — COMMENTED** [review]
**cron.ts:52** — the regex `(\d{1,2}:\d{2}\s*[AP]M)` assumes 12-hour format. `cronstrue` can return 24-hour time (e.g. `14:30`) in non-English locales, so the regex won't match and the "UTC" label silently disappears. Not a crash, just no annotation.

Consider matching 24h as well: `(\d{1,2}:\d{2}(\s*[AP]M)?)`.

**Comment by jsell-rh** [comment]
@machadovilaca thank you!

Review bot found one nit, and I've got one more request:

We're moving towards a spec-driven workflow for ambient. Could you use the `/spec` skill to create/update the specs to reflect this change?

We don't have full spec coverage at this point, but I do want to make sure this change is captured in the specs. 

Feel free to include those spec changes in this PR.

File: components/frontend/src/lib/cron.ts [hunk 3]
```diff
@@ -29,3 +30,32 @@ export function getNextRuns(cronExpr: string, count: number): Date[] {
...
+
+  const withUtc = description.replace(/(\d{1,2}:\d{2}\s*[AP]M)/, "$1 UTC");
+
+  const local = formatTimeLocal(runs[0]);
+
+  if (local.endsWith("UTC")) {
+    return withUtc;
+  }
+
+  return `${withUtc} (${local})`;
```
[the remark this answers is not in the document]
Comment (machadovilaca) [reply components/frontend/src/lib/cron.ts]: Now if the description has no recognizable time pattern, we bail out cleanly with just the plain description instead of appending a local time without a matching UTC label.

**Comment by jsell-rh** [comment]
LGTM

**Comment by jsell-rh** [comment]
@machadovilaca looks like lint might be failing

**Comment by jsell-rh** [comment]
The lint failure looks like it was introduced in #1544. Looking into a fix.

**Comment by jsell-rh** [comment]
Fixed lint in #1579
